.github/pull_request_template.md

```diff
@@ -0,0 +1,27 @@
+### What are you trying to accomplish?
+
+<!-- Provide both a what and a _why_ for the change. -->
+
+<!-- What issues does this affect or fix? -->
+
+### Anything you want to highlight for special attention from reviewers?
+
+<!-- If there were multiple ways to approach the problem, why did you pick this one? -->
+
+### How will you know you've accomplished your goal?
+
+<!--
+  * If you've reproduced an error, can you link to, or demonstrate the reproduction?
+  * If you've added a new feature, how will you demonstrate it to others?
+  * If you've refactored code, how will you demonstrate that the new code is functionally equivalent to the old code?
+-->
+
+### Checklist
+
+<!-- Before requesting review, please ensure that your pull request fulfills the following requirements: -->
+
+- [ ] I have run the complete test suite to ensure all tests and linters pass.
+- [ ] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
+- [ ] I have written clear and descriptive commit messages.
+- [ ] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
+- [ ] I have ensured that the code is well-documented and easy to understand.
```

Comment:
🤔 If we include these expectations in the pull request template, I think it makes sense to ensure they're in CONTRIBUTING.md. As far as I can tell, we're mostly covered 👇 

- ✅ Tests passing [is implied](https://github.com/dependabot/dependabot-core/blob/26239220c4a230356d687ac2ebd0d47195a0573f/CONTRIBUTING.md?plain=1#L19-L20)
- ✅ Adding tests is [explicitly listed](https://github.com/dependabot/dependabot-core/blob/26239220c4a230356d687ac2ebd0d47195a0573f/CONTRIBUTING.md?plain=1#L19)
- ✅ Commit messages are [well covered](https://github.com/dependabot/dependabot-core/blob/26239220c4a230356d687ac2ebd0d47195a0573f/CONTRIBUTING.md?plain=1#L27-L33)
- ✅ Description of the changes is in this template 🎉
- ⬜︎ I don't see anything about documentation or ease of understanding

I think it'd be fine to add this expectation in a followup PR, but IMO we should state this clearly in CONTRIBUTING.md if we expect it from contributors. What do you think?

Reply:
Good shout, done!